carWarGame.py

```diff
@@ -18,13 +20,13 @@ class Deck:
     suits = ['♣','♦','♥','♠']
 
     def __init__(self):
-       self.cards = self.generateDeck(Deck.ranks,Deck.suits)
+       self.cards = self.generateDeck()
 
 
-    def generateDeck(self,ranks, suits):
+    def generateDeck(self) -> list:
         deck = []
-        for suit in suits:
-            for rank in ranks:
+        for suit in Deck.suits:
+            for rank in Deck.ranks:
                 card = Card(rank,suit)
                 deck.append(card)
```

Comment:
```suggestion
        for rank, suit in zip(self.suits, self.ranks):
            deck.append(Card(rank, suit))
```

https://www.w3schools.com/python/ref_func_zip.asp

zip() might be a good idea to avoid 2 loops.

Reply:
I am unsure if zip() will work as intended because iterators do not have the same length, and the idea is to have all ranks by each suit.

Reply:
Agree with you.

what about list comprehension?

```
deck = [Card(rank, suit) for suit in self.suits for rank in self.ranks]
```

Reply:
Great idea! What if we leave it this way?

`return [Card(rank, suit) for suit in self.suits for rank in self.ranks]`

Reply:
Great idea!